src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -330,8 +330,6 @@ private static void AddProjectFileDependenciesForPackageReference(PackageSpec pr
                     .Flattened
                     .SingleOrDefault(library => library.Key.Name.Equals(project.Name, StringComparison.OrdinalIgnoreCase));
 
-                Debug.Assert(resolvedEntry != null, "Unable to find project entry in target graph, project references will not be added");
```

Comment:
I'm not a big fan of asserts, because I've rarely seen any value from them, but can you elaborate why you removed this?

Reply:
The Assert condition is possible in empty graphs case.   

https://github.com/NuGet/NuGet.Client/blob/df00340c27f8fb257bf6ec3eee5f60521d56e707/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs#L269
and
https://github.com/NuGet/NuGet.Client/blob/2abbaf45c812c04c6808ca788ed353ae035091ee/src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs#L325

targetGraph is null

and 
https://github.com/NuGet/NuGet.Client/blob/2abbaf45c812c04c6808ca788ed353ae035091ee/src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs#L329

resolvedEntry is null.